pkg/plugins/policies/matchers/match_test.go

```diff
@@ -0,0 +1,110 @@
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/model/rest"
+	_ "github.com/kumahq/kuma/pkg/plugins/policies"
+	"github.com/kumahq/kuma/pkg/plugins/policies/matchers"
+	policies_api "github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/api/v1alpha1"
+	test_matchers "github.com/kumahq/kuma/pkg/test/matchers"
+	"github.com/kumahq/kuma/pkg/util/yaml"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+)
+
+var _ = Describe("Match", func() {
+
+	readDPP := func(file string) *core_mesh.DataplaneResource {
+		dppYaml, err := os.ReadFile(path.Join("testdata", file))
+		Expect(err).ToNot(HaveOccurred())
+
+		dpp, err := rest.YAML.UnmarshalCore(dppYaml)
+		Expect(err).ToNot(HaveOccurred())
+		return dpp.(*core_mesh.DataplaneResource)
+	}
+
+	readResourceListResponse := func(file string) xds_context.Resources {
+		responseBytes, err := os.ReadFile(path.Join("testdata", file))
+		Expect(err).ToNot(HaveOccurred())
+
+		rawResources := yaml.SplitYAML(string(responseBytes))
+		resourceList := &policies_api.MeshTrafficPermissionResourceList{}
+		for _, rawResource := range rawResources {
+			resource, err := rest.YAML.UnmarshalCore([]byte(rawResource))
+			Expect(err).ToNot(HaveOccurred())
+			err = resourceList.AddItem(resource)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		return xds_context.Resources{
+			MeshLocalResources: map[core_model.ResourceType]core_model.ResourceList{
+				policies_api.MeshTrafficPermissionType: resourceList,
+			},
+		}
+	}
+
+	printPolicies := func(policies []core_model.Resource) string {
+		matchedPolicyList := &policies_api.MeshTrafficPermissionResourceList{}
+		for _, policy := range policies {
+			Expect(matchedPolicyList.AddItem(policy)).To(Succeed())
+		}
+		bytesBuffer := &bytes.Buffer{}
+		err := kubectl_output.NewPrinter().Print(rest.From.ResourceList(matchedPolicyList), bytesBuffer)
+		Expect(err).ToNot(HaveOccurred())
+		return bytesBuffer.String()
+	}
+
+	Describe("MatchedPolicies", func() {
+
+		type testCase struct {
+			dppFile                    string
+			resourceListResponseFile   string
+			matchedResourcesGoldenFile string
+		}
+
+		DescribeTable("should return a list of policies ordered by levels for the given DPP",
+			func(given testCase) {
+				// given DPP resource
+				dpp := readDPP(given.dppFile)
+
+				// given MeshTrafficPermissions
+				resources := readResourceListResponse(given.resourceListResponseFile)
+
+				// when
+				policies, err := matchers.MatchedPolicies(policies_api.MeshTrafficPermissionType, dpp, resources)
+				Expect(err).ToNot(HaveOccurred())
+
+				// then
+				Expect(printPolicies(policies.DataplanePolicies)).To(test_matchers.MatchGoldenYAML(path.Join("testdata", given.matchedResourcesGoldenFile)))
+			},
+			Entry("01. policies select the dataplane without collisions", testCase{
```

Comment:
How about instead of doing entries like this do a list. It's done this way there: https://github.com/kumahq/kuma/blob/e26dc6d0a3aae7909d4c16577d3bc6174bc97658/app/kumactl/cmd/install/install_control_plane_test.go#L79-L90

It makes it very simple to create a new test case

Reply:
but then you don't have a description of your entry like `01. policies select the dataplane without collisions`, I find it pretty useful

Reply:
I put this as a comment in the input `yaml`